<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>920</width>
    <height>551</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Search engine</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="1" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="7" column="0">
       <widget class="QLabel" name="requests_fieeld_line">
        <property name="text">
         <string>Requests</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0" colspan="4">
       <widget class="QListView" name="requests_line_list">
        <property name="editTriggers">
         <set>QAbstractItemView::SelectedClicked</set>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="config_path_line">
        <property name="text">
         <string>Path to Config.json</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0" colspan="5">
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="answers_field_line">
        <property name="text">
         <string>Results</string>
        </property>
       </widget>
      </item>
      <item row="10" column="4">
       <widget class="QPushButton" name="search_button">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Search</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="answers_path_line">
        <property name="text">
         <string>Path to Answers.json</string>
        </property>
       </widget>
      </item>
      <item row="8" column="4">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="requests_add_button">
          <property name="text">
           <string>Add...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="requests_delete_button">
          <property name="text">
           <string>Delete...</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="11" column="0" colspan="5">
       <widget class="QTreeWidget" name="results_tree_widget">
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="showDropIndicator" stdset="0">
         <bool>false</bool>
        </property>
        <property name="alternatingRowColors">
         <bool>false</bool>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::NoSelection</enum>
        </property>
        <property name="rootIsDecorated">
         <bool>true</bool>
        </property>
        <property name="columnCount">
         <number>3</number>
        </property>
        <attribute name="headerVisible">
         <bool>false</bool>
        </attribute>
        <column>
         <property name="text">
          <string notr="true">1</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string notr="true">2</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string notr="true">3</string>
         </property>
        </column>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="requests_path_line">
        <property name="text">
         <string>Path to Requests.json</string>
        </property>
       </widget>
      </item>
      <item row="7" column="3" colspan="2">
       <widget class="QComboBox" name="mode_select">
        <property name="currentText">
         <string>Standard</string>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <property name="maxVisibleItems">
         <number>5</number>
        </property>
        <property name="maxCount">
         <number>4</number>
        </property>
        <property name="insertPolicy">
         <enum>QComboBox::NoInsert</enum>
        </property>
        <item>
         <property name="text">
          <string>Standard</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Manual</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Do not use configurations' file</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Do not use requests' file</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QPushButton" name="config_path_button">
        <property name="text">
         <string>Open...</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="5">
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="12" column="4">
       <widget class="QPushButton" name="save_button">
        <property name="text">
         <string>Save</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1" colspan="2">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Engine's mode</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLineEdit" name="config_path_edit"/>
      </item>
      <item row="3" column="0">
       <widget class="QLineEdit" name="requests_path_edit"/>
      </item>
      <item row="5" column="0">
       <widget class="QLineEdit" name="answers_path_edit"/>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="no_requests_mark">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap>icons/warning_small.png</pixmap>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="no_config_mark">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap>icons/warning_small.png</pixmap>
        </property>
       </widget>
      </item>
      <item row="3" column="2" colspan="3">
       <widget class="QPushButton" name="requests_path_button">
        <property name="text">
         <string>Open...</string>
        </property>
       </widget>
      </item>
      <item row="5" column="2" colspan="3">
       <widget class="QPushButton" name="answers_path_button">
        <property name="text">
         <string>Open...</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2" colspan="2">
       <widget class="QPushButton" name="config_change">
        <property name="text">
         <string>Configure...</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <action name="actionOpen_Config_json">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Open Config.json...</string>
   </property>
   <property name="toolTip">
    <string>Open Config.json file</string>
   </property>
  </action>
  <action name="actionOpen_Requests_json">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Open Requests.json...</string>
   </property>
   <property name="toolTip">
    <string>Open Requests.json file</string>
   </property>
  </action>
  <action name="actionSave_Answers_json">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Save Answers.json...</string>
   </property>
   <property name="toolTip">
    <string>Choose path to Answers.json file</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset theme="window-close">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionabout">
   <property name="text">
    <string>About...</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>config_path_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>openConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>909</x>
     <y>52</y>
    </hint>
    <hint type="destinationlabel">
     <x>913</x>
     <y>33</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>requests_path_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>openRequests()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>909</x>
     <y>102</y>
    </hint>
    <hint type="destinationlabel">
     <x>916</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>answers_path_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>openAnswers()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>909</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>916</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>requests_add_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addRequests()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>908</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>916</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>requests_delete_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>deleteRequests()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>908</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>914</x>
     <y>283</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>config_path_edit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>modifyConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>364</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>requests_path_edit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>modifyRequests()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>517</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>533</x>
     <y>26</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>answers_path_edit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>modifyAnswers()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>646</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>677</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>config_change</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>modifyConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>826</x>
     <y>52</y>
    </hint>
    <hint type="destinationlabel">
     <x>839</x>
     <y>29</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>search_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>search()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>909</x>
     <y>367</y>
    </hint>
    <hint type="destinationlabel">
     <x>789</x>
     <y>385</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveResult()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>877</x>
     <y>528</y>
    </hint>
    <hint type="destinationlabel">
     <x>741</x>
     <y>529</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mode_select</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>849</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openConfig()</slot>
  <slot>openRequests()</slot>
  <slot>openAnswers()</slot>
  <slot>addRequests()</slot>
  <slot>deleteRequests()</slot>
  <slot>modifyConfig()</slot>
  <slot>modifyRequests()</slot>
  <slot>modifyAnswers()</slot>
  <slot>search()</slot>
  <slot>saveResult()</slot>
  <slot>changeMode()</slot>
 </slots>
</ui>
